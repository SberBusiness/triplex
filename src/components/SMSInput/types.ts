import React from 'react';
import {ESMSInputSize} from './enums';
import {TestProps} from '@sberbusiness/triplex/types/CoreTypes';

/** Свойства компонента SMSInput. */
export interface ISMSInputProps extends React.HTMLAttributes<HTMLDivElement>, TestProps {
    /** Значение кода. */
    code: string;
    /** Признак блокировки компонента. */
    disabled?: boolean;
    /** Признак наличия ошибки. */
    error?: boolean;
    /** Размер поля. */
    size: ESMSInputSize;
    /** Обработчик изменения кода. */
    onChangeCode: (code: string) => void;
    /** Обработчик отправки кода. */
    onSubmitCode: (code: string) => void;
}
