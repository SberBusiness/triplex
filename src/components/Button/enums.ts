/** Темы кнопки. */
export enum EButtonTheme {
    GENERAL = 'general',
    SECONDARY = 'secondary',
    DANGER = 'danger',
    DOTS = 'dots',
    LINK = 'link',
    TILE = 'tile',
}

/** Размеры кнопки. */
export enum EButtonSize {
    LG = 'LG',
    MD = 'MD',
    SM = 'SM',
}

/** Форма границы кнопки. */
export enum EButtonIconShape {
    SQUIRCLE = 'squircle',
    CIRCLE = 'circle',
}
