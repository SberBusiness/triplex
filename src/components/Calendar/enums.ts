/** Вариант выбора даты. */
export enum EPickType {
    datePick,
    monthYearPick,
}

/**
 * Направление переключения страницы календаря.
 */
export enum EChangePageDirection {
    FORWARD,
    BACKWARD,
}
