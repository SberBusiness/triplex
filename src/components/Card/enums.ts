/** Возможные размеры внутреннего отступа. */
export enum ECardContentPaddingSize {
    MD = 'md',
    SM = 'sm',
}

/** Возможные размеры скругления карточки. */
export enum ECardRoundingSize {
    MD = 'md',
    SM = 'sm',
}
