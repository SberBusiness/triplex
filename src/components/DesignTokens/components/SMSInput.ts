import {TDesignTokenValue, TDesignTokenValues} from '@sberbusiness/triplex/components/DesignTokens/types/DesignTokenTypes';

// Название токенов компонента SMSInput.
export const designTokensComponentsSMSInputKeys = ['Refresh_Fill_Empty', 'Refresh_Fill_Full'] as const;
// Тип, содержащий названия токенов компонента SMSInput.
export type TDesignTokensComponentsSMSInputKeys = (typeof designTokensComponentsSMSInputKeys)[number];
// Тип, содержащий названия токенов компонента SMSInput и их значения.
export type TDesignTokensComponentsSMSInputValue = Record<TDesignTokensComponentsSMSInputKeys, TDesignTokenValue>;
// Тип, содержащий названия токенов компонента SMSInput и их значения в светлой и темной теме.
export type TDesignTokensComponentsSMSInputValues = Record<TDesignTokensComponentsSMSInputKeys, TDesignTokenValues>;
// Тип локальных токенов компонента SMSInput.
export type TDesignTokensComponentsSMSInput = {SMSInput: TDesignTokensComponentsSMSInputValue};

// Токены компонента SMSInput в светлой и темной темах.
export const SMSInput_Tokens: TDesignTokensComponentsSMSInputValues = {
    Refresh_Fill_Empty: [{ref: 'Neutral.700'}, {ref: 'Basic.300'}], // var(--triplex-SMSInput-Refresh_Fill_Empty)
    Refresh_Fill_Full: [{ref: 'Basic.300'}, {ref: 'Neutral.700'}], // var(--triplex-SMSInput-Refresh_Fill_Full)
};
