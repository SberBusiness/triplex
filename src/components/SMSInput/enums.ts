/** Возможные размеры SMSInput. */
export enum ESMSInputSize {
    MD = 'md',
    SM = 'sm',
}
