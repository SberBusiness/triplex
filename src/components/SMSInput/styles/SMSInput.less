@import 'src/styles/style';

.globalSmsInput {
    display: inline-block;
    position: relative;
    width: 100%;

    .input {
        padding-left: 33px;
        padding-right: 33px;
        background-color: @color-snowy;
        border-radius: @radius-16;
        box-sizing: border-box;
        text-align: center;

        &.SM {
            height: 28px;
            padding-top: 5px;
            padding-bottom: 5px;
            border-radius: 14px;
        }

        &:focus {
            border-color: @color-graphite-01;
        }

        &.error {
            border-color: @color-ruby-02;
        }
    }

    .btnRefresh {
        position: absolute;
        left: 6px;
        outline: none;

        &.MD {
            top: 6px;
        }

        &.SM {
            top: 4px;
        }

        &.disabled {
            cursor: default;
        }

        .light {
            fill: @color-asphalt-02;
        }

        .dark {
            fill: @color-graphite-02;
        }
    }

    .btnSubmit {
        position: absolute;
        right: 6px;

        &.MD {
            top: 6px;
        }

        &.SM {
            top: 4px;
        }
    }
}
