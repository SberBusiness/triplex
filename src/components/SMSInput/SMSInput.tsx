import React, {useState} from 'react';
import {ISMSInputProps} from '@sberbusiness/triplex/components/SMSInput/types';
import {ESMSInputSize} from '@sberbusiness/triplex/components/SMSInput/enums';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';
import {SMSInputTooltip} from '@sberbusiness/triplex/components/SMSInput/components/SMSInputTooltip';
import {SMSInputSubmit} from '@sberbusiness/triplex/components/SMSInput/components/SMSInputSubmit';
import {SMSInputInput} from '@sberbusiness/triplex/components/SMSInput/components/SMSInputInput';
import {SMSInputRefresh} from '@sberbusiness/triplex/components/SMSInput/components/SMSInputRefresh';
import {SMSInputContext} from '@sberbusiness/triplex/components/SMSInput/SMSInputContext';

/** Возвращает CSS класс в зависимости от размера. */
const getSizeClassName = (size: ESMSInputSize) => {
    switch (size) {
        case ESMSInputSize.MD:
            return 'cssClass[MD]';
        case ESMSInputSize.SM:
            return 'cssClass[SM]';
    }
};

/** Внутренние составляющие SMSInput. */
interface ISMSInputComposition {
    Tooltip: typeof SMSInputTooltip;
    Refresh: typeof SMSInputRefresh;
    Input: typeof SMSInputInput;
    Submit: typeof SMSInputSubmit;
}

/** Компонент для ввода СМС. */
export const SMSInput: React.FC<ISMSInputProps> & ISMSInputComposition = (props) => {
    const {children, className, code, disabled, error, size, onChangeCode, onSubmitCode, ...htmlDivAttributes} = props;

    const [disabledSubmit, setDisabledSubmit] = useState(true);
    const [tooltipId, setTooltipId] = useState<string>();
    const classSMSInput = classnames('cssClass[globalSmsInput]', className);
    const sizeClassName = getSizeClassName(size);

    return (
        <SMSInputContext.Provider
            value={{
                code,
                disabled: !!disabled,
                disabledSubmit,
                error: !!error,
                onChangeCode,
                onSubmitCode,
                setDisabledSubmit,
                setTooltipId,
                sizeClassName,
                tooltipId,
            }}
        >
            <div className={classSMSInput} {...htmlDivAttributes}>
                {children}
            </div>
        </SMSInputContext.Provider>
    );
};

SMSInput.displayName = 'SMSInput';
SMSInput.Tooltip = SMSInputTooltip;
SMSInput.Refresh = SMSInputRefresh;
SMSInput.Input = SMSInputInput;
SMSInput.Submit = SMSInputSubmit;
