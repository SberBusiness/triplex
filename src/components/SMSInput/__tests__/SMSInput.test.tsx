import renderer from 'react-test-renderer';
import React from 'react';
import {SMSInput} from '../SMSInput';
import {ESMSInputSize} from '../enums';

jest.mock('@sberbusiness/icons/SmssignSrvxIcon20', () => ({
    SmssignSrvxIcon20: 'svg',
}));

jest.mock('@sberbusiness/icons/ClosetooltipSrvxIcon16', () => ({
    ClosetooltipSrvxIcon16: 'svg',
}));

jest.mock('@sberbusiness/icons/ClosenotificationSrvxIcon16', () => ({
    ClosenotificationSrvxIcon16: 'svg',
}));

jest.mock('react-dom', () => ({
    findDOMNode: () => ({addEventListener: jest.fn()}),
}));

describe('SMSInput', () => {
    beforeEach(() => {
        allure.feature('SMSInput');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <SMSInput data-test-id="SMSInput" code={''} size={ESMSInputSize.MD} onChangeCode={jest.fn()} onSubmitCode={jest.fn()}>
                    <SMSInput.Tooltip message="Запросить новый код">
                        <SMSInput.Refresh countdownTime={10} countdownTimeLeft={0} onRefresh={jest.fn()} />
                    </SMSInput.Tooltip>
                    <SMSInput.Input placeholder="Введите код" />
                    <SMSInput.Submit />
                </SMSInput>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
