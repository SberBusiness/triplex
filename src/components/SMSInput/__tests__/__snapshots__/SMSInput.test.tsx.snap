// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SMSInput renders correctly 1`] = `
<div
  className="cssClass[globalSmsInput]"
  data-test-id="SMSInput"
  data-tx="10.0.0"
>
  <button
    aria-disabled={false}
    className="cssClass[buttonIcon] cssClass[circle] hoverable cssClass[btnRefresh] cssClass[MD]"
    data-tx="10.0.0"
    onClick={[Function]}
    onKeyDown={[Function]}
    type="button"
  >
    <svg
      focusable="false"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <clipPath
        id="clipFront0.999"
      >
        <path
          d="M10 10L21.883 8.33A12 12 0 1 1 21.099 5.438Z"
        />
      </clipPath>
      <path
        className="cssClass[empty]"
        d="M18.1 9.101a.9.9 0 0 0-.9.9 7.245 7.245 0 1 1-1.998-4.95h-2.16a.9.9 0 1 0 0 1.8h4.077a.9.9 0 0 0 .9-.9v-4.05a.9.9 0 0 0-1.8 0v1.593A9 9 0 1 0 19 10.001a.9.9 0 0 0-.9-.9z"
      />
      <path
        className="cssClass[full]"
        clipPath="url(#clipFront0.999)"
        d="M18.1 9.101a.9.9 0 0 0-.9.9 7.245 7.245 0 1 1-1.998-4.95h-2.16a.9.9 0 1 0 0 1.8h4.077a.9.9 0 0 0 .9-.9v-4.05a.9.9 0 0 0-1.8 0v1.593A9 9 0 1 0 19 10.001a.9.9 0 0 0-.9-.9z"
      />
    </svg>
  </button>
  <input
    autoComplete="off"
    className="cssClass[input] cssClass[input] cssClass[MD]"
    data-tx="10.0.0"
    maxLength={8}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Введите код"
    value=""
  />
  <button
    className="cssClass[buttonIcon] cssClass[circle] hoverable cssClass[btnSubmit] cssClass[MD] disabled"
    data-tx="10.0.0"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    <svg />
  </button>
</div>
`;
